clash-lib/src/Clash/Core/Term.hs

```diff
@@ -173,6 +185,12 @@ type TmName     = Name Term
 -- | Binding in a LetRec construct
 type LetBinding = (Id, Term)
 
+-- TODO Should Eq be alpha equivalance?
+data Bind a
+  = NonRec Id a
+  | Rec [(Id, a)]
+  deriving (Eq, Show, Generic, NFData, Hashable, Binary, Functor)
```

Comment:
```suggestion
  deriving (Show, Generic, NFData, Hashable, Binary, Functor)
  
-- | Structural equivalence, not alpha equivalence
deriving instance Eq a => Eq (Bind a)
```

Reply:
I've left a comment by the `deriving` clause but otherwise left this. I think it would be a good idea to deal with structural / alpha equality for `Bind` in #1990, because I also have a similar thing with alpha-equivalence for patterns in #1976 that I want to move to that PR

Reply:
I don't want to add the `Structural` and `Alpha` newtypes suggested in #1987 to #1990 though, I want to do that in a separate PR. #1990 significantly reduces compile times which I don't want to delay merging just because there's a further improvement on the horizon, here's `master` vs #1990
```
christiaan@DESKTOP-O5QKI8I:~/clash-compiler$ time cabal run clash-testsuite -- -p VHDL.clash --hide-successes -j6
Up to date

All 378 tests passed (555.68s)

real    9m17.383s
user    54m37.413s
sys     2m58.446s

christiaan@DESKTOP-O5QKI8I:~/clash-compiler$ time cabal run clash-testsuite -- -p VHDL.clash --hide-successes -j6
Up to date

All 378 tests passed (410.44s)

real    6m51.604s
user    38m45.764s
sys     2m21.107s
```

Reply:
Right, but you add the `eqTerm` function, and I want to add `Bind` to that and remove the `TODO` (since I also want to dig out the alpha / structural equality on `Pat` and add it to that PR anyway we may as well do both)